### PR TITLE
config: rebalancer sharding role

### PR DESCRIPTION
**Pull request by ImeevMA**
This patchset introduces new sharding role - 'rebalancer'.

### CODE REVIEW

File: changelogs/unreleased/gh-8862-sharding-role-rebalancer.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+## feature/config
+
+* The sharding.role option has been disabled in the instance scope (gh-8862).
+* The 'rebalancer' sharding role was instroduced (gh-8862).
```
Comment (xuniq) [inline changelogs/unreleased/gh-8862-sharding-role-rebalancer.md:4]:
```suggestion
* Disabled the `sharding.role` option in the instance scope (gh-8862).
* Introduced the `rebalancer` sharding role (gh-8862).
```

**Review by xuniq — APPROVED** [review]

File: changelogs/unreleased/gh-8862-sharding-role-rebalancer.md (hunk 1 above)
Comment (ImeevMA) [reply changelogs/unreleased/gh-8862-sharding-role-rebalancer.md]: Thanks! Fixed. The day will come when I will write a changelog without errors. I hope ... :smile:

**Comment by ImeevMA** [comment]
Note that #9490 must be pushed before this commit.

**Review by ochaplashkin — APPROVED** [review]

File: src/box/lua/config/instance_config.lua [hunk 2]
```diff
@@ -1629,6 +1629,7 @@ return schema.new('instance_config', schema.record({
         roles = schema.set({
             'router',
             'storage',
+            'rebalancer',
```
Comment (Totktonada) [inline src/box/lua/config/instance_config.lua:1632]: Say, no rebalancer role assigned: does it mean that there is no rebalancer or that the rebalancer is chosen automatically? If the former, how to ask for the automation? If the latter, how to disable the rebalancer at all?

Comment (ImeevMA) [reply src/box/lua/config/instance_config.lua:1632]: For now, this means that the rebalancer will be selected from among all replicasets masters. Additionally, there is currently no option to disable the rebalancer via config.

Comment (Totktonada) [reply src/box/lua/config/instance_config.lua:1632]: It is OK for now, but I think that we should provide ability to completely disable the rebalancer for administration tasks later (system, tarantool or application upgrade; disaster recovery).

**Review by Totktonada — APPROVED** [review]

**Pull request merged**